src/scripts/zoom-controls.js

```diff
@@ -0,0 +1,227 @@
+    this.dollyDelta = new H5P.ThreeJS.Vector2();
+
+    // Register event listeners
+    element.addEventListener('wheel', this.onMouseWheel.bind(this), false);
+    element.addEventListener('touchstart', this.onTouchStart.bind(this), false);
+    element.addEventListener('touchmove', this.onTouchMove.bind(this), false);
+    element.addEventListener('keydown', this.onKeyDown.bind(this), false);
+  }
+
+  /**
+   * Get zoom scale.
+   * @returns {number} Zoom scale.
+   */
+  getZoomScale() {
+    return Math.pow(0.95, this.zoomSpeed);
+  }
+
+  isDollyInDisabled() {
+    return this.object.isPerspectiveCamera && this.object.fov <= this.minFov;
+  }
+
+  isDollyOutDisabled() {
+    return this.object.isPerspectiveCamera && this.object.fov >= this.maxFov;
+  }
+
+  /**
+   * Dollies in the camera e.g. zoom in.
+   * @param {number} dollyScale How much to dolly in.
+   */
+  dollyIn(dollyScale) {
+    if (dollyScale === undefined) {
+      dollyScale = this.getZoomScale();
+    }
+
+    if (this.object.isPerspectiveCamera) {
+      this.object.fov = Math.max(this.minFov, Math.min(this.maxFov, this.object.fov * dollyScale));
+      this.object.updateProjectionMatrix();
+    }
+    else if (this.object.isOrthographicCamera) {
+      this.object.zoom = Math.max(this.minZoom, Math.min(this.maxZoom, this.object.zoom * dollyScale));
+      this.object.updateProjectionMatrix();
+    }
+  }
+
+  /**
+   * Dollies out the camera e.g. zoom out.
+   * @param {number} dollyScale How much to dolly out.
+   */
+  dollyOut(dollyScale) {
+    if (dollyScale === undefined) {
+      dollyScale = this.getZoomScale();
+    }
+
+    if (this.object.isPerspectiveCamera) {
+      this.object.fov = Math.max(this.minFov, Math.min(this.maxFov, this.object.fov / dollyScale));
+      this.object.updateProjectionMatrix();
+    }
+    else if (this.object.isOrthographicCamera) {
+      this.object.zoom = Math.max(this.minZoom, Math.min(this.maxZoom, this.object.zoom / dollyScale));
+      this.object.updateProjectionMatrix();
+    }
+  }
+
+  /**
+   * Handle mouse wheel.
+   * @param {WheelEvent} event Mouse wheel event.
+   */
+  handleMouseWheel(event) {
+    if (event.deltaY < 0) {
+      this.dollyIn(this.getZoomScale());
+    } 
+    else if (event.deltaY > 0) {
+      this.dollyOut(this.getZoomScale());
+    }
+  }
+
+  /**
+   * Handle touch start.
+   * @param {TouchEvent} event Touch event.
+   */
+  handleTouchStartDolly(event) {
+    var dx = event.touches[0].pageX - event.touches[1].pageX;
+    var dy = event.touches[0].pageY - event.touches[1].pageY;
+
+    var distance = Math.sqrt(dx * dx + dy * dy);
+
+    this.dollyStart.set(0, distance);
+  }
+
+  /**
+   * Handle touch move.
+   * @param {TouchEvent} event Touch event.
+   */
+  handleTouchMoveDolly(event) {    
+    var dx = event.touches[0].pageX - event.touches[1].pageX;
+    var dy = event.touches[0].pageY - event.touches[1].pageY;
+
+    var distance = Math.sqrt(dx * dx + dy * dy);
```

Comment:
Let's use modern statements in this new code.

```suggestion
    const dx = event.touches[0].pageX - event.touches[1].pageX;
    const dy = event.touches[0].pageY - event.touches[1].pageY;

    const distance = Math.sqrt(dx * dx + dy * dy);
```